### ISSUE TITLE
Net::SSH::HostKeyMismatch with cert based host auth

### ISSUE THREAD

**Issue by thomas-merz** [theforeman/hammer-cli-foreman-ssh#10]
```
/Users/Thomas.Merz/.rvm/gems/ruby-2.7.6/gems/net-ssh-multi-1.2.1/lib/net/ssh/multi/pending_connection.rb:29:in `replay_on': undefined method `open_channel' for nil:NilClass (NoMethodError)
error connecting to admin@example.com: Net::SSH::HostKeyMismatch (fingerprint SHA256:xxxxx does not match for "example.com,172.12.34.56")
```
This has already been fixed in https://github.com/net-ssh/net-ssh/issues/613 but `hammer host ssh` doesn't honor this fix from August 2021 because there has been no updated build since April 2015. So please check this out and give us a newer version on https://rubygems.org/gems/hammer_cli_foreman_ssh

**Comment by thomas-merz** [comment]
Hey @dgoetz , can you help to give us Foremen a newer `hammer` version that doesn't have this issue, so that one can use it anymore because it's broken with more and more hosts over time 😞

**Comment by dgoetz** [comment]
No, this needs a new gem release. @ofedoren seems to be the only still active Foreman developer who contributed here, so I hope he can do a gem release, then I will pick it up for packaging on all supported versions.

**Comment by ekohl** [comment]
@thomas-merz a new release of this gem won't solve it.

Stack trace wise it points to:
https://github.com/net-ssh/net-ssh-multi/blob/f4c787ef3770f5218fefcb3c21091de13205ede8/lib/net/ssh/multi/pending_connection.rb#L29

So somehow `session` is `nil` there. This could very well be a bug in net-ssh-multi, but it's hard to say.

https://github.com/net-ssh/net-ssh-multi doesn't look maintained, so that could be a problem.

**Comment by thomas-merz** [comment]
Uh, yes, I can see [my very old issue from 2022](https://github.com/net-ssh/net-ssh-multi/issues/22) over there 😞
